### ISSUE TITLE
EKS ServiseAccount AWS_WEB_IDENTITY_TOKEN_FILE unix path notation on Windows nodes

### ISSUE THREAD

**Issue by ArturChe** [aws/amazon-eks-pod-identity-webhook#57]
**What happened**:
On Windows pod env variable looks like:
```
AWS_WEB_IDENTITY_TOKEN_FILE    /var/run/secrets/eks.amazonaws.com/serviceaccount/token
```

**What you expected to happen**:
```
AWS_WEB_IDENTITY_TOKEN_FILE    C:\var\run\secrets\eks.amazonaws.com\serviceaccount\token
```

**How to reproduce it (as minimally and precisely as possible)**:
Create the EKS cluster with windows nodes and configured Service Account with the proper IAM role.

**Anything else we need to know?**:
Thrown error from AWS SDK .Net:
```
The webIdentityTokenFile must be an absolute path.
```
AWSCLI v2 does works inside this pod and reacts to revoking policies from the role.

**Environment**:
- AWS Region: us-east-1
- EKS Platform version: "eks.1"
- Kubernetes version: "1.16"
- Webhook Version: eksctl 0.19.0

**Comment by micahhausler** [comment]
Did you update the `nodeSelector` on your pods as [described in the README](https://github.com/aws/amazon-eks-pod-identity-webhook#usage-with-windows-container-workloads)?

**Comment by ArturChe** [comment]
@micahhausler sure:
```
nodeSelector:
  kubernetes.io/arch: amd64
  kubernetes.io/os: windows
```

**Comment by ArturChe** [comment]
Accidentally closed

**Comment by ArturChe** [comment]
@mhausenblas Hi, are there any updates regarding the issue?

**Comment by mhausenblas** [comment]
Afraid, not from my side @ArturChe but maybe @micahhausler knows?

**Comment by saiteja313** [comment]
https://github.com/aws/amazon-eks-pod-identity-webhook/pull/26

**Comment by mcblair** [comment]
This is still an issue and is blocking us from moving into production.

Environment:

AWS Region: us-west-2
EKS Platform version: "eks.1"
Kubernetes version: "1.16"
Webhook Version: eksctl 0.21.0

```
nodeSelector:
  kubernetes.io/arch: amd64
  kubernetes.io/os: windows
```

**Comment by DylanMunyard** [comment]
I'm also having this problem on EKS 1.1.6 where the token file is a Unix path and I get the exception about the absolute path, source: https://github.com/aws/aws-sdk-net/blob/master/sdk/src/Core/Amazon.Runtime/Credentials/_bcl%2Bnetstandard/AssumeRoleWithWebIdentityCredentials.cs#L95

I've tried both beta and non-beta nodeSelector. How do I check the version of the pod identity web hook running in my cluster has the fix from #26? The same workaround as above does work, but means I'm making a code change to my app because when I try to update the AWS_WEB_IDENTITY_TOKEN_FILE env var in the entry script it keeps being reset.

**Comment by mcblair** [comment]
@DylanMunyard 

Is your app docker based? I'm not sure what your entrypoint is, but ours is a `.bat` that runs `set AWS_WEB_IDENTITY_TOKEN_FILE=C:\var\run\secrets\eks.amazonaws.com\serviceaccount\token` before executing the app.

**Comment by DylanMunyard** [comment]
@mcblair I did try that (mine is PowerShell), but I've just deleted the deployment and can confirm it works now:
`[System.Environment]::SetEnvironmentVariable("AWS_WEB_IDENTITY_TOKEN_FILE", "C:\var\run\secrets\eks.amazonaws.com\serviceaccount\token")`

I think what throws me off is that inside the pod when I run via PowerShell `$env:AWS_WEB_IDENTITY_TOKEN_FILE` it shows `/var/run/secrets/eks.amazonaws.com/serviceaccount/token` but inside the app the environment variable has the value I replaced in the entrypoint. 

Related to this issue - do you know anything about how AWS deploy the pod IAM web hook? It seems to be running 'magically' inside the cluster without a pod spec, so I'm thinking of building their image manually and deploying it to see if that works, not sure if you tried that already?

**Comment by DylanMunyard** [comment]
Hi @mcblair and @micahhausler I just wanted to let you know that when I deployed the pod identity webhook from the latest source in this repo that the path is set to C:\...

So it means the webhook running in EKS was not up to date. But i don't have visibility on how it was deployed as I never deployed it so I believe it came with the EKS cluster when I either deployed it or upgraded it. 

Do we expect it to be upgraded automatically, or need to raise a support case?

**Comment by mcblair** [comment]
@DylanMunyard You’re deploying `1.16` using `eksctl`?

**Comment by DylanMunyard** [comment]
Running EKS 1.16. In my cluster before I deployed with make cluster-up the pod identity web hook was running but the endpoint it served from was local host on a specific port so it feels like AWS deploy it themselves. After I deployed it changed the admission web hook to use a service instead that resolved to the image I built from latest source.

**Comment by ArturChe** [comment]
Hi all. The problem has been resolved in k8s v1.18. Update your EKS cluster.
